### ISSUE TITLE
How to report inside `Display` implementation?

### ISSUE THREAD

**Issue by kvark** [brendanzab/codespan#340]
This is useful for errors. If an error has some report, I'd like to be able to print it in the `Display` implementation.
However, I'm not able to find any way to do this. All the `term` based APIs require `std::io::Write`, while `Display` gives `std::fmt::Formatter : std::fmt::Write`.

See also - https://github.com/BurntSushi/termcolor/issues/55

**Comment by Johann150** [comment]
I don't think its easily possible to implement `std::fmt::Display` for `codespan_reporting::Diagnostic` because the source code text is not stored there. What would you expect from that `Display` implementation, could you give an example?

**Comment by kvark** [comment]
I didn't suggest implementing it for `Diagnostic`, no. It would be nice if `std::fmt::Formatter` implemented `WriteColor`.

**Comment by kvark** [comment]
Also, I found a way to output a string thanks to "Michael Lamparski" on matrix, so this can be closed now.
TL;DR: `Ansi<Vec<u8>>` implements `WriteColor`, then one can follow with `String::from_utf8_lossy`.
